transactions.go

```diff
@@ -58,9 +58,11 @@ func (t *Transactions) BatchAdd(block BlockID, transactions []Transaction, verif
 		filtered := transactions[:0]
 
 		for i := range transactions {
-			if transactions[i].VerifySignature() {
-				filtered = append(filtered, transactions[i])
+			if !transactions[i].VerifySignature() {
```

Comment:
Hm, could signature verification be moved to PrevalidateTransaction?

Reply:
Done.